### PR TITLE
nixos/oauth2-proxy{,-nginx}: renamed from oauth2_proxy, also renamed the servi…

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…ce, user, group

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 - `mkosi` was updated to v19. Parts of the user interface have changed. Consult the
   [release notes](https://github.com/systemd/mkosi/releases/tag/v19) for a list of changes.
 
+- `services.oauth2_proxy` was renamed to `services.oauth2-proxy`. Also the corresponding service, user and group where renamed.
```
Comment (AndersonTorres) [inline nixos/doc/manual/release-notes/rl-2405.section.md:35]:
```suggestion
- `services.oauth2_proxy` was renamed to `services.oauth2-proxy`. Also the corresponding service, user and group were renamed.
```

typo

File: nixos/modules/services/security/oauth2-proxy-nginx.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+let
+  cfg = config.services.oauth2-proxy.nginx;
+in
+{
+  options.services.oauth2-proxy.nginx = {
+    proxy = mkOption {
+      type = types.str;
+      default = config.services.oauth2-proxy.httpAddress;
+      defaultText = literalExpression "config.services.oauth2-proxy.httpAddress";
+      description = lib.mdDoc ''
```
Comment (AndersonTorres) [inline nixos/modules/services/security/oauth2-proxy-nginx.nix:12]: \> `with lib`
\> `lib.mdDoc`

This is why I don't like `with`.

**Review by AndersonTorres — APPROVED** [review]
Looks good to me.

File: nixos/modules/services/security/oauth2-proxy-nginx.nix (hunk 2 above)
Comment (SuperSandro2000) [reply nixos/modules/services/security/oauth2-proxy-nginx.nix:12]: That's from a treewide change. I don't think it really matters tbh.

File: nixos/modules/services/security/oauth2-proxy.nix [hunk 3]
```diff
@@ -547,35 +545,37 @@ in
...
-      example = "/run/keys/oauth2_proxy";
+      example = "/run/keys/oauth2-proxy";
     };
-
   };
 
-  config = mkIf cfg.enable {
+  imports = [
+    (mkRenamedOptionModule [ "services" "oauth2_proxy" ] [ "services" "oauth2-proxy" ])
+  ];
```
Comment (h7x4) [inline nixos/modules/services/security/oauth2-proxy.nix:554]: nonblocking:

It doesn't matter much, and I can't recall seeing any guidelines specifying the order of modules, but it feels a bit weird to put `imports` between `options` and `config`.

While looking through, I only found `nginx` that does it like this, and `postfix` which puts the imports on the bottom.

File: nixos/modules/services/security/oauth2-proxy-nginx.nix (hunk 2 above)
Comment (AndersonTorres) [reply nixos/modules/services/security/oauth2-proxy-nginx.nix:11]: Yes, but many "new code" repeats this nonsense.

Indeed in the short run it does not matter.

File: nixos/modules/services/security/oauth2-proxy.nix (hunk 3 above)
Comment (AndersonTorres) [reply nixos/modules/services/security/oauth2-proxy.nix:554]: Languages like C and Python put includes and imports are at the top.

Comment (SuperSandro2000) [reply nixos/modules/services/security/oauth2-proxy.nix:554]: This is pretty standard across the repo

**Comment by AndersonTorres** [comment]
Fix the conflicts

**Comment by SuperSandro2000** [comment]
rebased against master

**Comment by tadfisher** [comment]
@SuperSandro2000 Missed one import from `lib`, I think:

```
       error: undefined variable 'optionalAttrs'
       at /nix/store/3pif36ks3f56py4wb1dkq6sh0nkf3ygj-source/nixos/modules/services/security/oauth2-proxy.nix:20:47:
           19|
           20|     google = cfg: { google = with cfg.google; optionalAttrs (groups != []) {
             |                                               ^
           21|       admin-email = adminEmail;
```

**Comment by SuperSandro2000** [comment]
Didn't notice that because of the with usage 😅 

Was already fixed with #309611

**Pull request merged**